addons/web/static/src/views/fields/properties/property_definition.js

```diff
@@ -220,6 +226,12 @@ export class PropertyDefinition extends Component {
 
         delete propertyDefinition.comodel;
 
+        PropertyDefinition._propertyParametersMap.forEach((types, param) => {
+            if (!types.includes(propertyDefinition.type)) {
+                delete propertyDefinition[param];
+            }
+        });
+
```

Comment:
@std-odoo I make a fix to clean up parameters.
The behavior should be acceptable.
But when I read the code I think this method can be improved.
1. line 215 is the root of all problems since it may copy parameter data we don't want. logically, it should only copy those non-special parameters shared by all type e.g. ('name', 'string', 'definition_changed', 'view_in_cards'...) TBD: Using a white list might make everything simple, but we may miss adding them in the js when adding new parameters.
2. line 227 only delete `propertyDefinition.comodel` but not `propertyDefinition.domain` which means the rpc args can be `[{"name":"585969644e4b73f1","string":"Property 1","type":"many2many","default":false,"view_in_cards":true,"domain":"[('id', '<', 1000)]","value":false,"definition_changed":true}]}]` where we don't have `comodel` but have `domain`. (It happens when you have a many2one with comodel and domain and the type is changed to many2many
TBD: also delete `propertyDefinition.domain`?
3. when you change type to itself, for many2x type, the comodel will be reset, but for other type, e.g. selection, the selection will be kept. I am not really sure what is the expected behavior.